src/test/java/htsjdk/samtools/reference/SamLocusAndReferenceIteratorTest.java

```diff
@@ -53,4 +53,52 @@ public void testSamLocusAndReferenceIteratorMismatch() {
         final SamLocusIterator samLocusIterator = new SamLocusIterator(samReader);
         new SamLocusAndReferenceIterator(referenceSequenceFileWalker, samLocusIterator); // should throw
     }
+
+    @Test
+    public void testSamLocusAndReferenceIteratorLeadingInsertion() {
+        final File reference = new File(TEST_DATA_DIR, "Homo_sapiens_assembly18.trimmed.fasta");
+        final File samFile = new File(TEST_DATA_DIR, "leading_insertion.sam");
+        final ReferenceSequenceFile referenceSequenceFile = new FastaSequenceFile(reference, false);
+        final ReferenceSequenceFileWalker referenceSequenceFileWalker = new ReferenceSequenceFileWalker(referenceSequenceFile);
+
+        final SamReader samReader = SamReaderFactory.makeDefault().open(samFile);
+        final SamLocusIterator samLocusIterator = new SamLocusIterator(samReader);
+        samLocusIterator.setIncludeIndels(true);
+        final SamLocusAndReferenceIterator samLocusAndReferences = new SamLocusAndReferenceIterator(referenceSequenceFileWalker, samLocusIterator);
+
+        IntervalList intervalList = new IntervalList(samReader.getFileHeader());
+        intervalList.add(new Interval("chrM", 1, 26));
+        intervalList.add(new Interval("chrM", 16546, 16571));
+
+        OverlapDetector<Interval> overlapDetector = new OverlapDetector<>(0, 0);
+        overlapDetector.addAll(intervalList.getIntervals(), intervalList.getIntervals());
+
+        // Developer note: these are the # of loci that we have (1) a read base mapped to a reference base, (2) a read
+        // base inserted relative the reference, and (3) a read base deleted relative to the reference.  For read bases
+        // inserted prior to or after the reference (ex.. at position 1 of a contig), they are counted only once!
+        int totalMappedRecords = 0;
+        int totalInsertedRecords = 0;
+        int totalDeletedRecords = 0;
+        for (final SamLocusAndReferenceIterator.SAMLocusAndReference samLocusAndReference : samLocusAndReferences) {
+
```

Comment:
Could you also assert that you see reference position `0` and that the reference base is `N` at that position please?